### ISSUE TITLE
Change "Prefix Too Large" Behavior?

### ISSUE THREAD

**Issue by cooperwalbrun** [cooperwalbrun/aws-cidr-finder#4]
I will need to gather some input from others on this. The idea is that instead of erroring out when an "invalid" prefix is specified, e.g.:

```
$ aws-cidr-finder --profile myprofile --prefix 18
Desired prefix (18) is incompatible with the available CIDR blocks!
Encountered a CIDR whose prefix is 19, which is higher than 18. Offending CIDR: 172.31.96.0/19
Run the command again without the --prefix argument to see the full list.
```

...the results list could just be scrubbed of any CIDRs that do not satisfy the criteria. For example, if the `aws-cidr-finder` command would return `172.0.0.0/20, 172.0.16.0/18, 172.45.0.0/32` and we specify `--prefix 18`, the only result would be `172.0.16.0/18` because that is the only returned CIDR that is large enough to have the desired prefix.

**Comment by cooperwalbrun** [comment]
An alternative to changing the behavior of `--prefix` is adding an additional flag which will allow the behavior described above. What I have in mind is `--filter`, e.g.:

```bash
aws-cidr-finder --profile myprofile --prefix 18 --filter
```

**Comment by DacoDev** [comment]
> An alternative to changing the behavior of `--prefix` is adding an additional flag which will allow the behavior described above. What I have in mind is `--filter`, e.g.:
> 
> ```shell
> aws-cidr-finder --profile myprofile --prefix 18 --filter
> ```

Adding filter seems like a good solution, but also I don't know if someone looking for a subnet with a certain CIDR would  care if it runs into a block it checked that was of an incompatible size, only if there were no available blocks for their specified prefix. 

Of course, this is from my perspective and use-case where I'm breaking up a VPC to create a subnet with Terraform at launch time. I found this issue because I got the error you listed; the first available block sequentially in my VPC is  a /28, but I'm looking for /25s. 

Whatever you decide, thanks for making and working on this!
